hardware_interface/include/hardware_interface/system_interface.hpp

```diff
@@ -547,24 +547,77 @@ class SystemInterface : public rclcpp_lifecycle::node_interfaces::LifecycleNodeI
     lifecycle_state_ = new_state;
   }
 
-  void set_state(const std::string & interface_name, const double & value)
+  template <typename T>
+  void set_state(const std::string & interface_name, const T & value)
   {
-    system_states_.at(interface_name)->set_value(value);
+    auto it = system_states_.find(interface_name);
+    if (it == system_states_.end())
+    {
+      throw std::runtime_error(
+        "State interface not found: " + interface_name +
+        " in system hardware component: " + info_.name + ". This should not happen.");
+    }
+    auto & handle = it->second;
+    std::unique_lock<std::shared_mutex> lock(handle->get_mutex());
+    std::ignore = handle->set_value(lock, value);
```

Comment:
I'll just raise this now: if we want to ignore this, we should add the appropriate flags to the other side instead of duct taping here, no?